core/src/test/java/greencity/controller/TipsAndTricksControllerTest.java

```diff
@@ -0,0 +1,133 @@
+    }
+
+    @Test
+    public void saveTest() throws Exception {
+        Principal principal = Mockito.mock(Principal.class);
+        when(principal.getName()).thenReturn("");
+        String json = "{\n" +
+                "\"title\": \"title\",\n" +
+                " \"tags\": [\"news\"],\n" +
+                " \"text\": \"content content content\", \n" +
+                "\"source\": \"\",\n" +
+                " \"image\": null\n" +
+                "}";
+        MockMultipartFile jsonFile = new MockMultipartFile("tipsAndTricksDtoRequest", "", "application/json", json.getBytes());
+
+        when(tipsAndTricksService.save(any(TipsAndTricksDtoRequest.class), any(MultipartFile.class), anyString()))
+                .thenReturn(new TipsAndTricksDtoResponse());
+
+        this.mockMvc.perform(multipart(tipsAndTricksLink)
+                .file(jsonFile)
+                .principal(principal)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated());
+
+        verify(tipsAndTricksService, times(1))
+                .save(any(TipsAndTricksDtoRequest.class), any(), anyString());
+    }
+
+    @Test
+    public void saveBadRequestTest() throws Exception {
+        this.mockMvc.perform(post(tipsAndTricksLink)
+                .content("{}")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void getTipsAndTricksByIdTest() throws Exception {
+        when(tipsAndTricksService.findDtoById(anyLong()))
+                .thenReturn(new TipsAndTricksDtoResponse());
+
+        this.mockMvc.perform(get(tipsAndTricksLink + "/{id}", 1))
+                .andExpect(status().isOk());
+
+        verify(tipsAndTricksService, times(1))
+                .findDtoById(anyLong());
+    }
+
+    @Test
+    public void findAllTest() throws Exception {
+        this.mockMvc.perform(get(tipsAndTricksLink))
+                .andExpect(status().isOk());
+
+        verify(tipsAndTricksService, times(1))
+                .findAll(any());
+    }
+
+    @Test
+    public void deleteTest() throws Exception {
+        this.mockMvc.perform(delete(tipsAndTricksLink + "/{id}", 1))
+                .andExpect(status().isOk());
+
+        verify(tipsAndTricksService, times(1))
+                .delete(anyLong());
+    }
+
+    @Test
+    public void getTipsAndTricksTest() throws Exception {
+        this.mockMvc.perform(get(tipsAndTricksLink + "/tags"))
+                .andExpect(status().isOk());
+
+        verify(tipsAndTricksService, times(1))
+                .find(any(), any());
+    }
+
+    @Test
+    public void findAllTipsAndTricksTagsTest() throws Exception {
+        this.mockMvc.perform(get(tipsAndTricksLink + "/tags/all"))
+                .andExpect(status().isOk());
+
+        verify(tagService, times(1))
```

Comment:
You can skip `times(1)`, it the default.

Reply:
removed times(1)